mainapp/static/mainapp/model_stats.js

```diff
@@ -0,0 +1,153 @@
+        function (error) {
+            console.error("Error loading model:", error);
+                document.getElementById("model-status").textContent = "Error loading model: " + error.message;
+                document.getElementById("model-status").style.display = "block";
+                document.getElementById("model-preview").style.display = "none";
+        }
+    );
+}
+
+function processModel(gltf) {
+    const stats = calculateModelStats(gltf.scene, gltf.animations);
+    updateModelStats(stats);
+}
+
+function calculateModelStats(model, animations) {
+    let faceCount = 0;
+    let meshCount = 0;
+    let materialCount = 0;
+    const seenMaterials = new Set();
+    const boundingBox = new THREE.Box3().setFromObject(model);
+
+    model.traverse(function (child) {
+        if (child.isMesh) {
+            meshCount++;
+        }
+        if (child.isMesh && child.geometry) {
+            if (child.geometry.index) {
+                faceCount += child.geometry.index.count / 3;
+            } else if (child.geometry.attributes.position) {
+                faceCount += child.geometry.attributes.position.count / 3;
+            }
+        }
+        if (child.isMesh && child.material){
+            const materials = Array.isArray(child.material)
+                ? child.material
+                : [child.material];
+            materials.forEach((mat) => {
+                if (!seenMaterials.has(mat.uuid)) {
+                    seenMaterials.add(mat.uuid);
+                    materialCount += 1;
+                }
+            });
+        }
+    });
+
+    const size = boundingBox.getSize(new THREE.Vector3());
+
+    // we will need to enforce scale=1
+    const scale = model.scale;
+    const hasAnimations = Array.isArray(animations) && animations.length > 0;
+    const triangleDensity = faceCount / (size.x * size.y * size.z || 1);
+
+    return {
+        faces: Math.floor(faceCount),
+        volume: size.x * size.y * size.z,
+        triangleDensity: triangleDensity.toFixed(2),
+
+        boundingBox: {
+            width: size.x.toFixed(2),
+            height: size.y.toFixed(2),
+            depth: size.z.toFixed(2)
+        },
+        scale: {
+            x: scale.x.toFixed(2),
+            y: scale.y.toFixed(2),
+            z: scale.z.toFixed(2)
+        },
+
+        materials: materialCount,
+        meshes: meshCount,
+        hasAnimations,
+    };
+}
+
+function setStatQualityClass(id, quality) {
+    const el = document.getElementById(id);
+    if (!el) return;
+
+    const classList = ["stat-good", "stat-warning", "stat-bad", "stat-neutral"];
+    el.classList.remove(...classList);
+
+    if (quality && classList.includes("stat-" + quality)) {
+        el.classList.add("stat-" + quality);
+    }
+}
+
+function updateModelStats(stats) {
+    const fc = stats.faces;
+    document.getElementById("faceCount").textContent = fc.toLocaleString();
+    setStatQualityClass("faceCount", 
+        fc >= 200 ? "good" :
```

Comment:
We mostly warn about face counts which are too _high_. Low face counts can be ok with suitable texturing (or for objects with simple geometry), so they're not necessarily an issue.